docs/data/base/components/badge/badge.md

```diff
@@ -1,43 +1,149 @@
+## Introduction
+
+A badge is a small descriptor for UI elements.
+It typically sits on or near an element and indicates the status of that element by displaying a number, icon, or other short set of characters.
+
+The `BadgeUnstyled` component creates a badge that is applied to its child element.
+
+{{"component": "modules/components/ComponentLinkHeader.js", "design": false}}
+
+## Component
+
+### Anatomy
+
+After [installation](/base/getting-started/installation/), you can start building with this component using the following basic elements:
+
+```jsx
 import BadgeUnstyled from '@mui/base/BadgeUnstyled';
+
+export default function MyApp() {
+  return (
+    <BadgeUnstyled>{/* the element that the badge is attached to */}</BadgeUnstyled>
+  );
+}
 ```
 
-{{"component": "modules/components/ComponentLinkHeader.js", "design": false}}
+### Basic usage
+
+`BadgeUnstyled` wraps around the UI element that it's attached to.
+For instance, if the badge indicates the number of emails in an inbox, then the component will be structured like this:
+
+```jsx
+<BadgeUnstyled>
+  <MailIcon />
+</BadgeUnstyled>
+```
+
+### Component slots
+
+The `BadgeUnstyled` component is composed of a root `<span>` that houses the element that the badge is attached to, followed by a `<span>` slot to represent the badge itself:
+
+```html
+<span class="BaseBadge-root">
+  <!-- the element the badge is attached to is nested here -->
+  <span class="BaseBadge-badge">badge content</span>
+</span>
+```
+
+### Slot props
+
+:::info
+The following props are available on all non-utility Base components.
+See [Usage](/base/getting-started/usage/) for full details.
+:::
+
+Use the `component` prop to override the root slot with a custom element:
+
+```jsx
+<BadgeUnstyled component="div" />
+```
+
+Use the `components` prop to override any interior slots in addition to the root:
+
+```jsx
+<BadgeUnstyled components={{ Root: 'div', Badge: 'div' }} />
+```
+
+:::warning
+If the root element is customized with both the `component` and `components` props, then `component` will take precedence.
+:::
+
+Use the `componentsProps` prop to pass custom props to internal slots.
+The following code snippet applies a CSS class called `my-badge` to the badge slot:
+
+```jsx
+<BadgeUnstyled componentsProps={{ badge: { className: 'my-badge' } }} />
+```
+
+:::warning
+Note that `componentsProps` slot names are written in lowercase (`root`) while `components` slot names are capitalized (`Root`).
+:::
+
+## Hook
+
+```jsx
+import { useBadge } from '@mui/base/BadgeUnstyled';
+```
+
+The `useBadge` hook lets you apply the functionality of `BadgeUnstyled` to a fully custom component.
+It returns props to be placed on the custom component, along with fields representing the component's internal state.
+
+Hooks _do not_ support [slot props](#slot-props), but they do support [customization props](#customization).
+
+:::info
+Hooks give you the most room for customization, but require more work to implement.
```

Comment:
I recently realized another use case for hooks - if you're building a component library that can be customized further by developers (e.g. using the `componentsProps` prop), use hooks. Unstyled components are better suited to build "closed" libraries (that's why we use hooks to build Joy components, for example).

Reply:
Do you think this is worth mentioning on all pages? Or maybe it would be better suited for the Usage page?

Reply:
I suppose having it just on the Usage page should be enough.

Reply:
Sounds good. I'll add it to #33272